alfresco-integration/shared/src/main/java/com/github/dynamicextensionsalfresco/models/M2ModelResourceListProvider.java

```diff
@@ -0,0 +1,67 @@
+package com.github.dynamicextensionsalfresco.models;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.alfresco.error.AlfrescoRuntimeException;
+import org.alfresco.repo.dictionary.M2Model;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ResourceLoaderAware;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.util.Assert;
+
+public class M2ModelResourceListProvider implements M2ModelListProvider, ResourceLoaderAware {
+
+    private Logger logger = LoggerFactory.getLogger(M2ModelResourceListProvider.class);
+
+    private final String modelLocationPattern = "osgibundle:/META-INF/alfresco/models/*.xml";
+
+    protected ResourcePatternResolver resourcePatternResolver;
+
+    @Override
+    public void setResourceLoader(ResourceLoader resourceLoader) {
+        Assert.isInstanceOf(ResourcePatternResolver.class, resourceLoader);
+        this.resourcePatternResolver = (ResourcePatternResolver) resourceLoader;
+    }
+
+
+    private List<M2ModelResource> _models;
+
+    @Override
+    public List<M2ModelResource> getModels() {
+        if (this._models == null) {
```

Comment:
This is the best I could come up with for "lazy initialization" without introducing new dependencies.

apache-commons-lang has a LazyInitializer<T> - but only since 3.0 - and we still support 2.x with Aflresco 5.0